test/e2e/specs/editor/various/preview.spec.js

```diff
@@ -0,0 +1,362 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	previewUtils: async ( { page }, use ) => {
+		await use( new PreviewUtils( { page } ) );
+	},
+} );
+
+test.describe( 'Preview', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should open a preview window for a new post', async ( {
+		editor,
+		page,
+		pageUtils,
+		previewUtils,
+		requestUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Disabled until content present.
+		const isPreviewDisabled = await editorPage
+			.locator( 'role=button[name="Preview"]' )
+			.isDisabled();
+		expect( isPreviewDisabled ).toEqual( true );
+
+		await editorPage.type(
+			'role=textbox[name="Add title"]',
+			'Hello World'
+		);
+
+		const previewPage = await editor.openPreviewPage( editorPage );
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+
+		expect( await previewTitle.textContent() ).toEqual( 'Hello World' );
+
+		// When autosave completes for a new post, the URL of the editor should
+		// update to include the ID. Use this to assert on preview URL.
+		const [ , postId ] = await (
+			await editorPage.waitForFunction( () => {
+				return window.location.search.match( /[\?&]post=(\d+)/ );
+			} )
+		 ).jsonValue();
+
+		const expectedPreviewURL = new URL( requestUtils.baseURL );
+		expectedPreviewURL.search = `?p=${ postId }&preview=true`;
+		expect( previewPage.url() ).toEqual( expectedPreviewURL.href );
+
+		// Return to editor to change title.
+		await editorPage.bringToFront();
+		await editorPage.type( 'role=textbox[name="Add title"]', '!' );
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		expect( await previewTitle.textContent() ).toEqual( 'Hello World!' );
+
+		// Pressing preview without changes should bring same preview window to
+		// front and reload, but should not show interstitial.
+		await editorPage.bringToFront();
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		expect( await previewTitle.textContent() ).toEqual( 'Hello World!' );
+
+		// Preview for published post (no unsaved changes) directs to canonical URL for post.
+		await editorPage.bringToFront();
+		await editor.publishPost();
+
+		// Close the panel.
+		await page.locator( 'role=button[name="Close panel"]' ).click();
+
+		// Return to editor to change title.
+		await editorPage.bringToFront();
+		await editorPage.waitForSelector( 'role=textbox[name="Add title"]' );
+		await editorPage.click( 'role=textbox[name="Add title"]' );
```

Comment:
Playwright will ensure the stability of the element when clicking on it, so I don't think we have to wait for it?

```suggestion
		await editorPage.click( 'role=textbox[name="Add title"i]' );
```

Reply:
Must have overlooked that one. I already removed a bunch of `waitForSelector`s from the original test 😄